File: java/src/main/java/mx/iteso/Deleter.java
package mx.iteso;

import mx.iteso.PrivilegeDecorator;
import mx.iteso.User;

public class Deleter extends PrivilegeDecorator {
    private User user;

    public Deleter(User user) {
        this.user = user;
    }

    @Override
    public String getDescription() {
        return this.user.getDescription() + " Can delete";
    }

    @Override
    public boolean publish() {
        return this.user.publish();
    }

    @Override
    public boolean delete() {
        return true;
    }

    @Override
    public boolean approve() {
        return this.user.approve();
    }

    @Override
    public boolean comment() {
        return this.user.comment();
    }
}

File: python/src/main/python/Deleter.py
class Deleter():
    def __init__(self, oldUser):
        self.user = oldUser

        def getDescription(self):
            return self.user.getDescription() + " Can delete"

        def publish(self):
            return self.user.publish()

        def delete(self):
            return True

        def approve(self):
            return self.user.approve()

        def comment(self):
            return self.user.comment()
